micromanager_gui/main_window.py

```diff
@@ -231,13 +475,16 @@ def _refresh_camera_options(self):
 
     def _refresh_objective_options(self):
         if "Objective" in self._mmc.getLoadedDevices():
+            self.objective_comboBox.clear()
             self.objective_comboBox.addItems(self._mmc.getStateLabels("Objective"))
 
     def _refresh_channel_list(self):
         if "Channel" in self._mmc.getAvailableConfigGroups():
+            self.snap_channel_comboBox.clear()
+            self.mda.clear_channel()
+            self.explorer.clear_channel()
```

Comment:
same thing here... can the event that should trigger self.explorer to clear it's channels, be connected in that widget, rather than here?

Reply:
@tlambert03 I moved everything related to the explorer widget into sample_explorer.py (move all explorer features in explorer widget: https://github.com/tlambert03/napari-micromanager/pull/17/commits/22e3419e34bc2d041069459677054783179bdb28)

Reply:
looks good thanks.  but something has broken in the meantime: the `test_main_window` test is now failing... any ideas?

Reply:
I've been trying to understand why but I'm not sure...it seems that with the current settings in `test_main_window` there won't be any layer in the viewer...
Can it be because `SEQUENCE_META` is not defined in the tests?
```
def _on_mda_frame(self, image: np.ndarray, event: useq.MDAEvent):
        seq = event.sequence
        meta = self.mda.SEQUENCE_META.get(seq, {})

        if meta.get("mode") == "mda":
          ...
```

Reply:
not sure yet, but I can say that it happened in 22e3419 (697c7e2e8c5440475339e1c4ab5c492aa9a5f168 was fine)

Reply:
@tlambert03 Ok I think is fixed. The test was missing the `SEQUENCE_META` info.